libs/ng-strapi-auth/src/lib/core/interceptor/http.auth-interceptor.ts

```diff
@@ -0,0 +1,27 @@
+
+import { HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http'
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+
+@Injectable()
+export class HttpAuthInterceptor implements HttpInterceptor {
+
+    constructor(private readonly route: Router) { }
+
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        const token = sessionStorage.get('currentJwt');
+        let newHeaders = req.headers.set('Authorization', `Bearer ${token}`);
+        const httpReq = req.clone({
+            headers: newHeaders,
+        })
+        return next.handle(httpReq).pipe(
+            catchError((error: HttpErrorResponse) => {
+                this.route.navigate(['/authentication'])
+                throw error.error;
+            })
```

Comment:
on any request error you logout the user ?... if we log him out, we should use a logout function from the services to do it properly and do it only on 401 / 403 responses ...